build-info-extractor-maven3/src/main/java/org/jfrog/build/extractor/maven/ArtifactoryProjectBuilder.java

```diff
@@ -52,53 +50,16 @@ public List<ProjectBuildingResult> build(List<File> pomFiles, boolean recursive,
 
     private List<ArtifactRepository> getRepositories() {
         List<ArtifactRepository> repositories = new ArrayList<>();
+        ProxySelector proxySelector = new ProxySelector(resolutionHelper.getHttpProxyHost(), resolutionHelper.getHttpProxyPort(), resolutionHelper.getHttpProxyUsername(), resolutionHelper.getHttpProxyPassword(), resolutionHelper.getHttpsProxyHost(), resolutionHelper.getHttpsProxyPort(), resolutionHelper.getHttpsProxyUsername(), resolutionHelper.getHttpsProxyPassword(), resolutionHelper.getNoProxy());
 
-        String releaseRepoUrl = resolutionHelper.getRepoReleaseUrl();
-        String snapshotRepoUrl = resolutionHelper.getRepoSnapshotUrl();
-
-        org.apache.maven.artifact.repository.Authentication authentication = null;
-        if (StringUtils.isNotBlank(resolutionHelper.getRepoUsername())) {
-            authentication = new org.apache.maven.artifact.repository.Authentication(resolutionHelper.getRepoUsername(), resolutionHelper.getRepoPassword());
-        }
-        org.apache.maven.repository.Proxy proxy = null;
-        if (StringUtils.isNotBlank(resolutionHelper.getProxyHost())) {
-            proxy = new org.apache.maven.repository.Proxy();
-            proxy.setHost(resolutionHelper.getProxyHost());
-            proxy.setPort(resolutionHelper.getProxyPort());
-            proxy.setUserName(resolutionHelper.getProxyUsername());
-            proxy.setPassword(resolutionHelper.getProxyPassword());
+        ArtifactoryPluginResolution artifactoryResolution = new ArtifactoryPluginResolution(resolutionHelper.getRepoReleaseUrl(), resolutionHelper.getRepoSnapshotUrl(), resolutionHelper.getRepoUsername(), resolutionHelper.getRepoPassword(), proxySelector, new NullPlexusLog());
```

Comment:
Providing a null logger may hide some valuable information. Can we use a regular log or use an adapter to our logger?

Reply:
Removed`NullPlexusLog()` in flavor of `resolutionHelper.getLogger()`